xapian-core/api/valuerangeproc.cc

```diff
@@ -432,8 +432,120 @@ NumberRangeProcessor::operator()(const string& b, const string& e)
     }
 
     return RangeProcessor::operator()(
-	    b.empty() ? b : Xapian::sortable_serialise(num_b),
-	    e.empty() ? e : Xapian::sortable_serialise(num_e));
+	b.empty() ? b : Xapian::sortable_serialise(num_b),
+	e.empty() ? e : Xapian::sortable_serialise(num_e));
+
+not_our_range:
+    return Xapian::Query(Xapian::Query::OP_INVALID);
+}
+
+static const unsigned int size_map[26] = {
+    0,
+    1, // B
+    0, 0, 0, 0,
+    1024 * 1024 * 1024, // G
+    0, 0, 0,
+    1024, // K
+    0,
+    1024 * 1024, // M
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+};
+
+static unsigned int
+get_size_multiplier_from_suffix(const char ch)
```

Comment:
The `const` here doesn't really server a useful purpose.  Generally you only want to make a function parameter `const` if it's a pointer or reference.